src/System.Private.ServiceModel/tests/Scenarios/Binding/Custom/CustomBindingTests.cs

```diff
@@ -51,4 +51,69 @@ public static void DefaultSettings_Tcp_Binary_Echo_RoundTrips_String()
             ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
         }
     }
+
+    public class MessageModifier : ChannelMessageInterceptor
+    {
```

Comment:
Test files should only contain test methods, could you move this to a separate file or into one of the other test support files you already have.

I have a task to clean this up in other areas of our test bed.